data/blog/2024/2024-09-26-using-laminas-continuous-delivery-and-deployment.md

```diff
@@ -0,0 +1,61 @@
+---
+id: 2024-09-26-using-laminas-continuous-delivery-and-deployment
+author: julian
+title: 'Using Laminas Continuous Delivery and Deployment'
+draft: false
+public: true
+created: '2024-09-26T11:00:00-01:00'
+updated: '2024-09-26T11:00:00-01:00'
+tags:
+  - continuous integration
+---
+
+We detailed the meaning and some inner workings of Continuous Integration (**CI**) in a [previous article](https://getlaminas.org/blog/2024-08-05-using-laminas-continuous-integration.md.html), so here we will focus on what happens after CI completes successfully.
+
+CI is followed by **Continuous Delivery** (**CD**) which gets the code **ready for deployment to the production or staging environment**.
+This ensures that the code is ready to be released **manually**.
+**Continuous Deployment** goes one step further and **automates the release** itself.
+The development team has to decide on what CD is right for them, but in both cases the end result is **more frequent and reliable software updates**.
+
+<!--- EXTENDED -->
+
+### Continuous Delivery vs. Continuous Deployment
+
+Continuous Delivery ensures that the updates you have been working hard on are ready to be deployed to the production or staging environment.
+This means that your code has already passed a battery of unit, integration and system tests.
+GitHub Actions can automate this process by having you define the steps that follow right before you get to the release itself, which is still performed manually.
+Continuous Deployment adds the final step by performing the release automatically.
+
+### How does Laminas Continuous Deployment work?
+
+Let's investigate a **practical example** for creating a release:
+
+- **Push your updates** onto a separate branch, to avoid conflicts with other developers working on the same code.
+- Optionally, **merge the update** into a 'developer' branch, which can result in conflicts.
+- Merge the 'developer' branch into the 'main' branch. Note the branch setup and names may differ.
+- Optionally, **update your CHANGELOG.md** file.
+- **Create a new release tag** which can follow the {MAJOR}.{MINOR}.{BUGFIX} versioning format.
+- **Push the release.**
+- **Create the release** on GitHub using the new tag and the description from the CHANGELOG.md file.
+
+**Laminas creates a milestone** for each new release that may include several issues and pull requests.
```

Comment:
Creates "Laminas" something? I don't think so! 😉 
And at this point, it is assumed that everyone knows what milestones on GitHub are. I think that should be concretised. This can help: https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones